### ISSUE TITLE
Linter confused by active pattern

### ISSUE THREAD

**Issue by tpetricek** [fsprojects/VisualFSharpPowerTools#1244]
I'm not suggesting anybody should actually write this sort of code :-), but:

![lint](https://cloud.githubusercontent.com/assets/485413/11104248/62d4169c-88bf-11e5-942f-6e70d9708319.png)

The F# compiler warning ("Incomplete pattern matches ...") makes sense and that's fine.

But the linter warning which says "Expected a camel case identifier, found 'None'." is wrong, because the `None` appears in a location where a pattern is expected - and so it is not a variable but a reference to standard `None` DU case.

**Comment by duckmatt** [comment]
@vasily-kirichenko @dungpa is it be possible for the linter to be able to use the typed AST or will that be too much of a performance killer? At the moment there are a few cases that require knowledge of the types for an accurate hint

**Comment by vasily-kirichenko** [comment]
Yes, we can easily pass typed ast to it, but the linter tags will be updating with longer delay during code editing (same as colorization).

**Comment by duckmatt** [comment]
Once I've finished the current bugs and performance upgrades I'm currently working on I'll try to take a look at at least making it optional to type check - perhaps returning an observable list so the hints are updated as the lint progresses, and see how that performs
